### PR TITLE
[#381] Add colcon package for iceoryx2

### LINKED ISSUE #381: Colcon package for C/C++ bindings
## Brief feature description

Add a `package.xml` to enable building `iceoryx2` C or C++ bindings with colcon.

## Detailed information

Enables `iceoryx2` usage in `ros2` projects that are built with `colcon`.

### PR DESCRIPTION
**Pull request by orecham**
## Notes for Reviewer

As per title

## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked in the [References](#references) section
1. [ ] ~~Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`~~
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [ ] ~~Tests follow the [best practice for testing][testing]~~
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [x] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

Closes #381

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <name>iceoryx2_cxx</name>
+    <version>0.3.0</version>
+    <description>iceoryx2 with c/c++ bindings</description>
+    <maintainer email="iceoryx-oss-support@ekxide.io">ekxide developers</maintainer>
+    <license>Apache 2.0</license>
+    <url type="bugtracker">https://github.com/eclipse-iceoryx/iceoryx2/issues</url>
+    <url type="repository">https://github.com/eclipse-iceoryx/iceoryx2</url>
+
+    <buildtool_depend>cmake</buildtool_depend>
+    <depend>iceoryx_hoofs</depend>
```
Comment (elBoberido) [inline package.xml:13]: Doesn't this cause build issues?

**Review by elBoberido — COMMENTED** [review]
LGTM, just one question to resolve

File: package.xml (hunk 1 above)
Comment (orecham) [reply package.xml]: @elBoberido If the right version of `iceoryx_hoofs` is in the colcon workspace it works. 

Nevertheless, I have added a version attribute to the dependency which will print a warning if the version is too low. Doesn't fail the build outright though with this reasoning (prints other errors that are a result of the version mismatch along with the warning.

Comment (elBoberido) [reply package.xml]: Right, this will only be a problem once we become part of the ros2.repos list

**Review by elBoberido — APPROVED** [review]

**Review by elBoberido — APPROVED** [review]

**Pull request merged**